src/hotspot/os/windows/os_windows.cpp

```diff
@@ -5990,6 +5990,32 @@ bool os::win32::find_mapping(address addr, mapping_info_t* mi) {
   return rc;
 }
 
+// Helper for print_one_mapping: print n words, both as hex and ascii.
+// Use Safefetch for all values.
+static void print_snippet(const void* p, outputStream* st) {
+  static const int num_words = LP64_ONLY(3) NOT_LP64(6);
+  static const int num_bytes = num_words * sizeof(int);
+  intptr_t v[num_words];
+  const int errval = 0xDE210244;
+  for (int i = 0; i < num_words; i++) {
+    v[i] = SafeFetch32((int*)p + i, errval);
+    if (v[i] == errval) {
```

Comment:
How about
```
if (SafeFetch32((int*)p + i, 0x1234) == 0x1234 && 
    SafeFetch32((int*)p + i, 0x5678) == 0x5678) {
  return;
} else {
  v[i] = SafeFetch32((int*)p + i, 0);
}

Reply:
Okay, I'll do that (minus the else part). I left it out because come on, its a 1 in 4 billion chance :) but the second SafeFetch only triggers when the first condition is true so this may still be cleaner.